core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -902,22 +902,29 @@ public void deleteTenant(String tenantUniqueID) throws UserStoreException {
         if (tenant != null) {
             clearTenantCaches(tenant);
             invalidateCacheManager(tenant.getDomain());
-        }
-        String sqlStmt = TenantConstants.DELETE_TENANT_BY_UUID_SQL;
-        try (Connection dbConnection = getDBConnection()) {
-            try (PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
-                prepStmt.setString(1, tenantUniqueID);
-                prepStmt.executeUpdate();
-                dbConnection.commit();
-            } catch (SQLException e) {
-                DatabaseUtil.rollBack(dbConnection);
+        } else {
+            String msg = String.format("Tenant not found for  tenant unique ID %s.", tenantUniqueID);
+            if (log.isDebugEnabled()) {
+                log.debug(msg);
             }
+            throw new UserStoreException(msg);
+        }
+        Connection dbConnection = null;
+        try {
+            dbConnection = getDBConnection();
```

Comment:
use try-with-resources

Reply:
We cannot use try with resources for db connection because two sql queries are executed for this db connection. So if we use try with resources, the rollback event will not work properly.

Reply:
fixed 7a2993d891c6d43a6a02ac1f9897c06ed4d4c735